libtcmu_log.c

```diff
@@ -179,8 +180,13 @@ log_internal(int pri, struct tcmu_device *dev, const char *funcname,
 	head = logbuf->head;
 	rb_set_pri(logbuf, head, pri);
 	msg = rb_get_msg(logbuf, head);
-	n = sprintf(msg, "%s:%d %s: ", funcname, linenr,
-		    dev ? dev->tcm_dev_name: "");
+	if (dev || rhandler) {
```

Comment:
I do not like the other patch, because it will just hide real bugs. I think here you just need to code the checks.

```

If (dev) {
    handler = tcmu_get_dev_handler();

    .....

```

Note you should use tcmu_get_dev_handler because some of the other callers are from comsumer and synth which are not runner based.

Reply:
Nevermind on the dev_handler comment. I am guessing the ltcmulib_handler->name is too verbose.

I guess since we are saying we do not care about the libtcmu abstraction anymore, do:
```

If (dev) {
    rhandler = tcmu_get_runner_handler();

```